tests/system/conftest.py

```diff
@@ -29,12 +32,91 @@ def pytest_runtest_makereport(item: Function, call: CallInfo) -> TestReport:
     result: TestReport = outcome.get_result()
 
     try:
-        post_report_to_slack(result, os.getenv("MLRUN_SYSTEM_TESTS_SLACK_WEBHOOK_URL"))
+        post_report_failed_to_slack(
+            result, os.getenv("MLRUN_SYSTEM_TESTS_SLACK_WEBHOOK_URL")
+        )
     except Exception as exc:
         print(f"Failed to post test report to slack: {exc}")
 
 
-def post_report_to_slack(report: TestReport, slack_webhook_url: str):
+def pytest_sessionstart(session):
+
+    # caching test results
+    session.results = collections.defaultdict(TestReport)
+
+
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    outcome = yield
+    report = outcome.get_result()
+
+    # cache test call result
+    if report.when == "call":
+        item.session.results[item] = report
+
+
+def pytest_sessionfinish(session: Session, exitstatus: ExitCode):
+    slack_url = os.getenv("MLRUN_SYSTEM_TESTS_SLACK_WEBHOOK_URL")
+    if slack_url:
+        post_report_session_finish_to_slack(
+            session,
+            exitstatus,
+            slack_url,
+        )
+
+
+def post_report_session_finish_to_slack(
+    session: Session, exitstatus: ExitCode, slack_webhook_url
+):
+    mlrun_version = os.getenv("MLRUN_VERSION", "")
+    mlrun_system_tests_component = os.getenv("MLRUN_SYSTEM_TESTS_COMPONENT", "")
+    ttl_tests = session.testscollected
+    ttl_failed_tests = session.testsfailed
```

Comment:
not sure I understand the `ttl` relation to the tests

Reply:
tl;dr how many (total). changed the variable name on below commit